pkg/ingester/client/streaming.go

```diff
@@ -90,6 +90,7 @@ func (s *SeriesChunksStreamReader) StartBuffering() {
 		onError := func(err error) {
 			s.errorChan <- err
 			log.Error(err)
+			level.Error(log).Log("msg", "received error while streaming chunks from ingester", "err", err)
```

Comment:
Isn't a double log?

Reply:
[`SpanLogger.Error()`](https://github.com/grafana/mimir/blob/79b9eadcbaf6ed33b6e0ee3e3cadb5ac038e1de6/vendor/github.com/grafana/dskit/spanlogger/spanlogger.go#L138) (line 92) sets the error flag on the span and logs the error as an event on the span.

[`SpanLogger.Log()`](https://github.com/grafana/mimir/blob/79b9eadcbaf6ed33b6e0ee3e3cadb5ac038e1de6/vendor/github.com/grafana/dskit/spanlogger/spanlogger.go#L108) (line 93) logs to the span (as an event on the span) and the application logs.

So with this change, yes, we'll end up with two events on the span with the error, but that seemed like the best option: I considered changing `SpanLogger.Error()` to log to the application logs, but my concern was potentially dramatically increasing the volume of application logs in cases where the existing behaviour (only adding an event to the span) was desirable. 

I'm open to other opinions / ideas though. Another option could be adding another method to `SpanLogger` that sets the error flag on the span and logs to both destinations.